collagraph/cgx/cgx.py

```diff
@@ -0,0 +1,196 @@
+from html.parser import HTMLParser
+
+from collagraph import create_element
+from collagraph.components import Component, ComponentMeta
+
+
+SUFFIX = "cgx"
+
+
+def load(path):
+    """
+    Loads and returns a component from a CGX file.
+
+    A subclass of Component will be created from the CGX file
+    where the contents of the <template> tag will be used as
+    the `render` function, while the contents of the <script>
+    tag will be used to provide the rest of the functions of
+    the component.
+
+    For example:
+
+        <template>
+          <item foo="bar">
+            <item baz="bla"/>
+          </item>
+        </template
+
+        <script>
+        import collagraph as cg
+
+        class Foo(cg.Component):
+            pass
+        </script>
+
+    """
+    parser = CGXParser()
+    parser.feed(path.read_text())
+
+    # TODO: proper validation of parsed structure
+
+    # Read the data from script block
+    script = parser.root.child_with_tag("script").data
+    # Construct a render function from the template
+    template_root_node = parser.root.child_with_tag("template").children[0]
+
+    def render(self):
+        return convert_node(template_root_node, self)
+
+    # Exec the script with a custom locals dict to capture
+    # all the defined classes and methods
+    local_attrs = {}
+    ComponentMeta.RENDER_FUNCTION = render
+    try:
+        exec(script, globals(), local_attrs)
```

Comment:
If I'm reading the docs right, it would be better to only pass in a globals dict.

If you pass both a globals and locals dict, the code is evaluated as though it were embedded in a class definition and I don't think you want that.

Reply:
> Remember that at module level, globals and locals are the same dictionary. If exec gets two separate objects as globals and locals, the code will be executed as if it were embedded in a class definition.

Yeah, I need this to be executed at module level, so I shouldn't pass in a locals object. Thanks!